prefect_sqlalchemy/credentials.py

```diff
@@ -253,3 +260,811 @@ def dict(self, *args, **kwargs) -> Dict:
             self.rendered_url.render_as_string(hide_password=False)
         )
         return d
+
+
+class SqlAlchemyUrl(BaseModel):
```

Comment:
I think this name is a bit more descriptive.
```suggestion
class SqlAlchemyConnectionParameters(BaseModel):
```

Also, I think there's an opportunity to put a method on this model that returns a fully formed connection URL.

Reply:
I dropped the SqlAlchemy prefix and renamed Parameters with Components since that's what sqlalchemy refers them as:
```

class URL(
    util.namedtuple(
        "URL",
        [
            "drivername",
            "username",
            "password",
            "host",
            "port",
            "database",
            "query",
        ],
    )
):
    """
    Represent the components of a URL used to connect to a database.
    ```